extensions/amp-byside-placeholder/0.1/amp-byside-placeholder.js

```diff
@@ -0,0 +1,348 @@
+      iframe.setAttribute('width', width);
+    }
+    if (this.element.hasAttribute('height')) {
+      iframe.setAttribute('height', height);
+    }
+
+    setStyles(iframe, {
+      'opacity': 0,
+    });
+
+    const self = this;
+
+    return this.generateSrcUrl_().then(src => {
+      this.iframeSrc_ = src;
+      iframe.src = this.iframeSrc_;
+
+      listenFor(iframe, 'embed-size', data => {
+        this.updateSize_(data['height'], data['width']);
+      });
+
+      this.applyFillContent(iframe);
+      this.element.appendChild(iframe);
+    }).then(() => {
+      this.iframePromise_ = this.loadPromise(iframe).then(() => {
+        this.getVsync().mutate(() => {
+          setStyles(iframe, {
+            'opacity': 1,
+            'width': width + 'px',
+            'height': height + 'px',
+          });
+        });
+      });
+    }).then(() => self);
+  }
+
+  /** @private */
+  generateOrigin_() {
+    const subDomain = this.webcareZone_ === MAIN_WEBCARE_ZONE_ ?
+      MAIN_WEBCARE_ZONE_SUBDOMAIN_ :
+      this.webcareZone_;
+    return 'https://' + subDomain + '.byside.com';
+  }
+
+  /** @private */
+  generateSrcUrl_() {
+    const src = this.baseUrl_ + 'placeholder.php';
+    const params = dict();
+
+    params['label'] = this.label_;
+    params['webcare_id'] = this.webcareId_;
+    params['bwch'] = this.channel_ || '';
+    params['lang'] = this.lang_ || '';
+    params['fid'] = this.fid_ || '';
+    params['bwit'] = (this.fid_ ? 'I' : 'A');
+    params['tuid'] = 'CLIENT_ID(byside_webcare_tuid)';
+    params['suid'] = '';
+    params['puid'] = 'PAGE_VIEW_IDpTIMESTAMP';
+    params['referrer'] = 'DOCUMENT_REFERRER';
+    params['page'] = 'SOURCE_URL';
+    params['amppage'] = 'AMPDOC_URL';
+    params['bwpt'] = 'TITLE';
+    params['bres'] = 'VIEWPORT_WIDTHxVIEWPORT_HEIGHT';
+    params['res'] = 'SCREEN_WIDTHxSCREEN_HEIGHT';
+    params['v'] = 'v20171116a';
+    params['ampv'] = 'AMP_VERSION';
+    params['viewer'] = 'VIEWER';
+    params['ua'] = 'USER_AGENT';
+    params['r'] = 'RANDOM';
+
+    if (this.isResizable_) {
+      params['_resize'] = 1;
+    }
+
+    return Services.urlReplacementsForDoc(this.element)
+        .expandAsync(addParamsToUrl(src, params));
+  }
+
+  /**
+   * Updates the element's dimensions to accommodate the iframe's
+   *    requested dimensions.
+   * @param {number|undefined} height
+   * @param {number|undefined} width
+   * @private
+   */
+  updateSize_(height, width) {
+    if (!this.isResizable_) {
+      this.user().error(TAG_,
+          'Ignoring embed-size request because this iframe is not resizable',
+          this.element);
+      return;
+    }
+
+    // Calculate new width and height of the container to include the padding.
+    // If padding is negative, just use the requested width and height directly.
+    let newHeight, newWidth;
+    height = parseInt(height, 10);
+    if (!isNaN(height)) {
+      newHeight = Math.max(
+          height + (this.element./*OK*/offsetHeight
```

Comment:
element and iframe should always have the same offset width/height as `this.applyFillContent(iframe)` enforces iframe to fill the component. I am not sure in what situation is code is needed. Could you clarify?